agent/engine/engine_unix_integ_test.go

```diff
@@ -1331,3 +1394,55 @@ func TestTaskLevelVolume(t *testing.T) {
 	client := taskEngine.(*DockerTaskEngine).client
 	client.RemoveVolume(context.TODO(), "TestTaskLevelVolume", 5*time.Second)
 }
+
+func TestGPUAssociationTask(t *testing.T) {
+	gpuSupportEnabled := utils.ParseBool(getGPUEnvVar(gpuConfigFilePath)["ECS_ENABLE_GPU_SUPPORT"], false)
```

Comment:
Grabbing this is the better approach here! I'd rather use the existing code to do it, though:

```golang
metadataClient := ec2.NewEC2MetadataClient(nil)
iid, _ := metadataClient.InstanceIdentityDocument()
// NewConfig already has the logic to parse the ecs.config file
fileCfg, err := config.NewConfig(metadataClient)
if err != nil {
    t.Skip("Couldn't load gpu config")
}
gpuSupportEnabled := fileCfg.GPUSupportEnabled
```

Reply:
ecs_config.go from the path /var/lib/ecs/ecs.config.go is getting read in init and then agent is started with environment variable ECS_ENABLE_GPU_SUPPORT= true/false . So that is why NewConfig wont be able to read from that path in integration test if we use NewConfig to set the gpuSupportEnabled.